File: magistere/1erannee/projet_tutore/cristofell.m
%Met la dimention ici 
n = 4;

x = sym("x", [1 n])

%les constantes/variables
G = sym("G");
M = sym("M");
c = sym("c");
Rs = sym("Rs");

%Rentre la métrique ici
g = [
    -(1-Rs/x(2)), 0, 0, 0;
    0, 1/(1-Rs/x(2)), 0, 0;
    0, 0, x(2)^2, 0;
    0, 0, 0, x(2)^2 * sin(x(3))^2;
]

ig = inv(g)

Gamma = g;

for i = 1:n
    Gamma(:,:,i) = zeros(n,n);
end

for gamma = 1:n
    for alpha = 1:n
        for beta = 1:n
            tmp0 = 0;
            for m = 1:n
                tmp0 = tmp0 + 0.5 * ig(gamma, m) * (diff(g(m,alpha), x(beta)) + diff(g(m, beta), x(alpha)) - diff(g(alpha, beta), x(m)));
            end
            Gamma(alpha, beta, gamma) = simplify(tmp0);
        end
    end
end

Gamma


gd = det(g)



Ri = g;

for mu = 1:n
    for nu = 1:n
        tmp1 = 0;
        tmp2 = 0;
        tmp3 = 0;
        tmp4 = 0;
        for alpha = 1:n
            tmp1 = tmp1 + diff(Gamma(mu, nu, alpha), x(alpha));
            tmp3 = tmp3 + Gamma(mu, nu, alpha) * diff(log(sqrt(-gd)), x(alpha));
            for beta = 1:n
                tmp2 = tmp2 + Gamma(alpha, mu, beta) * Gamma(beta, nu, alpha);
            end
        end
        Ri(mu,nu) = simplify(- diff(diff(log(sqrt(-gd)),x(nu)), x(mu)) + tmp1 - tmp2 + tmp3);
    end
end

Ri

courbureScalaire = 0;
for mu = 1:n
    for nu = 1:n
        courbureScalaire = courbureScalaire + ig(mu, nu) * Ri(mu, nu);
    end
end

courbureScalaire

File: magistere/1erannee/projet_tutore/compte_rendu/cristofell.cxx
X:=copy([t,r,theta,phi]);
d:=len(X);
g:=copy([[1-Rs/r,0,0,0],[0,-(1-Rs/r)^-1,0,0],[0,0,-r^2,0],[0,0,0,(-r^2)*sin(theta)^2]]);
ginv:=g^-1;
G:=(makemat(matrix(d),0,d))[0];
for (j:=0;j<d;j++) {for (k:=0;k<d;k++) {for (l:=0;l<d;l++) { 
        tmp:=0;  
        for (m:=0;m<d;m++) {tmp=(tmp+1/2*(ginv[j])[m]*(diff((g[k])[m],X[l])+diff((g[l])[m],X[k])-(diff((g[k])[l],X[m]))))};  
        ((G[j])[k])[l]:=simplify(tmp);  
      }}};
dg:=det(g);
Ri:=makemat(d);
for (j:=0;j<d;j++) {for (k:=0;k<d;k++) { 
      tmp1:=0;  
      tmp2:=0;  
      tmp3:=0;  
      tmp4:=0;  
      for (a:=0;a<d;a++) { 
          tmp1:=tmp1+diff(((G[a])[j])[k],X[a]);  
          tmp3:=tmp3+((G[a])[j])[k]*diff(ln(sqrt(-dg)),X[a]);  
          for (b:=0;b<d;b++) {tmp2:=tmp2+((G[b])[a])[j]*((G[a])[b])[k]};  
        };  
      (Ri[j])[k]:=simplify(-(diff(diff(ln(sqrt(-dg)),X[k]),X[j]))+tmp1-tmp2+tmp3);  
    }};
R:=0;
for (j:=0;j<d;j++) {for (k:=0;k<d;k++) {R=(R+(ginv[j])[k]*(Ri[j])[k])}};
